model-optimizer/extensions/middle/ShapeOfConstFolding.py

```diff
@@ -0,0 +1,38 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from mo.front.subgraph_matcher import SubgraphMatch
+from mo.graph.graph import Graph, rename_nodes
+from mo.middle.replacement import MiddleReplacementPattern
+from mo.ops.const import Const
+
+
+class ShapeOfConstFolding(MiddleReplacementPattern):
+    """
+    The transformation calculates the value of the constant
+    Const() -> ShapeOf() --> Const()
+    """
+    enabled = True
+
+    def pattern(self):
+        return dict(
+            nodes=[
+                ('const', dict(op='Const')),
+                ('const_data', dict(kind='data')),
+                ('shape', dict(op='ShapeOf')),
+            ],
+            edges=[
+                ('const', 'const_data', {}),
+                ('const_data', 'shape', {})
+            ])
+
+    def replace_pattern(self, graph: Graph, match: [dict, SubgraphMatch]):
+        shape = match['shape']
+
+        shape_name = shape.soft_get('name', shape.id)
+
+        shape_value = shape.in_port(0).data.get_value()
```

Comment:
the code doesn't work again. it is the shape that should become the value, not the constant value.